### PR TITLE
gh-142594: fix by property calls io.TextIOWrapper.detach

### PR DESCRIPTION
**Pull request by yihong0618**
add `self->detached` check 

* Issue: gh-142594

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2025-12-14-18-30-48.gh-issue-142594.belDmD.rst
Comment (StanFromIreland) [inline Misc/NEWS.d/next/Library/2025-12-14-18-30-48.gh-issue-142594.belDmD.rst:1]: Please move this entry to the library section.

Comment (yihong0618) [reply Misc/NEWS.d/next/Library/2025-12-14-18-30-48.gh-issue-142594.belDmD.rst:1]: copy that

**Review by cmaloney — APPROVED** [review]
LGTM

File: Misc/NEWS.d/next/Library/2025-12-14-18-30-48.gh-issue-142594.belDmD.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Fix crash in ``TextIOWrapper.close()`` when the underlying buffer's
+``closed`` property calls ``TextIOWrapper.detach()``.
```
Comment (cmaloney) [inline Misc/NEWS.d/next/Library/2025-12-14-18-30-48.gh-issue-142594.belDmD.rst]: nit: 

```suggestion
``closed`` property calls :meth:`~io.TextIOBase.detach`.
```

Comment (yihong0618) [reply Misc/NEWS.d/next/Library/2025-12-14-18-30-48.gh-issue-142594.belDmD.rst]: Thanks and done

**Review by serhiy-storchaka — APPROVED** [review]
LGTM. :+1:

**Comment by cmaloney** [comment]
(Working on the backports; auto failed because of my refactor / move of `test_io` tests)
